packages/medusa-file-local/src/services/local-file-service.ts

```diff
@@ -29,40 +31,91 @@ class LocalService extends AbstractFileService implements IFileService {
   async uploadFile(
     file: Express.Multer.File,
     options = {}
-  ): Promise<{ url: string }> {
-    const parsedFilename = parse(file.originalname)
+  ): Promise<{ url: string; key: string }> {
+    const parsedFilename = path.parse(file.originalname)
 
-    const fileKey = `${parsedFilename.name}-${Date.now()}${parsedFilename.ext}`
+    if (parsedFilename.dir) {
+      this.ensureDirExists(parsedFilename.dir)
+    }
+
+    const fileKey = path.join(
+      parsedFilename.dir,
+      `${Date.now()}-${parsedFilename.base}`
+    )
 
     return new Promise((resolve, reject) => {
       fs.copyFile(file.path, `${this.uploadDir_}/${fileKey}`, (err) => {
         if (err) {
+          reject(err)
           throw err
         }
 
         const fileUrl = `${this.backendUrl_}/${this.uploadDir_}/${fileKey}`
 
-        resolve({ url: fileUrl })
+        resolve({ url: fileUrl, key: fileKey })
       })
     })
   }
 
   async delete(file): Promise<void> {
-    throw Error("Not implemented")
+    const filePath = `${this.uploadDir_}/${file.fileKey}`
+    if (fs.existsSync(filePath)) {
+      fs.unlinkSync(filePath)
+    }
   }
 
   async getUploadStreamDescriptor(
-    fileData
+    fileData: UploadStreamDescriptorType
   ): Promise<FileServiceGetUploadStreamResult> {
-    throw Error("Not implemented")
+    const parsedFilename = path.parse(
+      fileData.name + (fileData.ext ? `.${fileData.ext}` : "")
+    )
+
+    if (parsedFilename.dir) {
+      this.ensureDirExists(parsedFilename.dir)
+    }
+
+    const fileKey = path.join(
+      parsedFilename.dir,
+      `${Date.now()}-${parsedFilename.base}`
```

Comment:
self-review: import strategy can't find ops files because we prepend date to the filename

Reply:
Can you add a bit more detail to this comment?

Reply:
This is resolved, import strategy was building a deterministic filekey for the ops files but the local file service adds a timestamp to the name which was preventing the strategy from correctly fetching the file. Now we persist file keys of the ops files on the batch job.